### PR TITLE
Bug 1109641 - Disable the back/forward buttons when there is nothing to do 

### PR DESCRIPTION
**Pull request by apbendi**
Taking a stab at how communication between the Browser and the Toolbar might work. All I'm doing now is changing enabled/disabled status of the back/forward buttons, but this obviously sets up for displaying load status, stop/reload button, etc...

No hard feelings if rejected, but would love some feedback on the approach! Cheers.

### CODE REVIEW

File: Client/Frontend/Browser/Browser.swift [hunk 1]
```diff
@@ -31,6 +43,10 @@ class Browser {
     var canGoForward: Bool {
         return webView.canGoForward
     }
+    
+    var estimatedProgresss: Float {
```
Comment (thebnich) [inline Client/Frontend/Browser/Browser.swift]: Progresss -> Progress

**Comment by thebnich** [comment]
See this thread for a discussion on this abstraction: https://mail.mozilla.org/pipermail/mobile-firefox-dev/2014-December/001010.html

Since we're building a full web browser, we might as well expose the entire WebView since we'll likely hook into nearly every method in the API anyway. The idea is that BVC is already so tightly coupled to the WebView that it might as well have access to it directly. But we intend to (manually) limit the number of consumers that access the WebView -- currently just BrowserViewController -- to prevent leaking the WebView API everywhere.

Other components, like the toolbar, need only know about a basic Tab interface (with a URL, title, back/forward state, etc.). If we can figure out a clean way to prevent these components from accessing the WebView at all (perhaps a more restricted Tab interface that Browser implements), I think we will; for now, we're simply making sure we don't touch the WebView outside of BVC.

Also, apologize for the late response to this PR. Most of us have been on vacation for the past couple of weeks!

**Comment by thebnich** [comment]
Closing this since it will be fixed by #59.

**Comment by apbendi** [comment]
Cool, thanks for pointing me in the direction of the thread. Makes sense! Appreciate you taking the time to review. And don't worry, I was on vacation too! Happy New Year =)

**Pull request closed without merging**